tests/lua/cartesi-machine-tests.lua

```diff
@@ -730,33 +740,47 @@ local function run_parallel(contexts)
                 ctx.cycles = ctx.target.run(machine, ctx, 2 * ctx.expected_cycles)
                 check_test_result(ctx)
                 ctx.target.destroy(machine)
-                unistd._exit(0)
+                if ctx.failed then
+                    unistd._exit(3)
+                else
+                    unistd._exit(0)
+                end
             else -- parent
-                pids[pid] = true
+                pids[pid] = ctx
                 running_jobs = running_jobs + 1
             end
         end
         while running_jobs >= jobs do
             -- wait a child to finish
             local pid, reason, exitcode = syswait.wait(-1)
             if pid and pid > 0 and reason ~= "running" then
-                if not (reason == "exited" and exitcode == 0) then add_error("unexpected child process exit") end
+                if exitcode == 3 then
```

Comment:
It should only interpret `exitcode` as an actual exit code in case `reason` is `"exited"`, otherwise this number is a signal code for `"killed"` or `"stopped"` reason and has different meaning. This was being considered before. But not a big issue since this will also work, code 3 is `SIGQUIT` when reason is `"killed"`, the CI may kill the process in case of a crash or out of memory, and it won't be with code 3.

Reply:
Removed, will be correctly fixed on another PR